src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,258 @@
+package Duke;
+
+import Duke.Exception.InvalidTaskException;
+import Duke.Exception.MissingDescriptonException;
+import Duke.Exception.MissingDoneNumberFieldException;
+import Duke.Exception.MissingTimeFieldException;
+import Duke.TaskTypes.Deadline;
+import Duke.TaskTypes.Event;
+import Duke.TaskTypes.Task;
+import Duke.TaskTypes.Todo;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    public static final int MAX_NO_OF_TASKS = 100;
+    public static final String TODO = "todo";
+    public static final String DEADLINE = "deadline";
+    public static final String EVENT = "event";
+    public static final String LIST = "list";
+    public static final String DONE = "done";
+    public static final int INITIAL_NO_OF_TASKS = 0;
+
+    public static void main(String[] args) {
+        displayHello();
+        Task[] taskList = initializeTaskList();
+        Scanner myInput = initializeScanner();
+        String userInput = getUserInput(myInput);
+        int numberOfTasks = INITIAL_NO_OF_TASKS;
+        while (isNotBye(userInput)) {
+            String[] splitUserInput = splitTheUserInput(userInput);
+            numberOfTasks = doTaskAndGetNewNumberOfTasks(taskList, numberOfTasks, userInput, splitUserInput);
+            userInput = getNextUserInput(myInput);
+        }
+        displayGoodbye();
+        closeScanner(myInput);
+    }
+
+    private static boolean isNotBye(String userInput) {
+        return !userInput.equals("bye");
+    }
+
+    private static String getUserInput(Scanner myInput) {
+        return myInput.nextLine();
+    }
+
+    private static String getNextUserInput(Scanner myInput) {
+        System.out.println("What else do you want to do?");
+        return myInput.nextLine();
+    }
+
+    private static String[] splitTheUserInput(String userInput) {
+        System.out.println(System.lineSeparator());
+        return userInput.split(" ");
+    }
+
+    private static void closeScanner(Scanner myInput) {
+        myInput.close();
+    }
+
+    private static Scanner initializeScanner() {
+        return new Scanner(System.in);
+    }
+
+    private static Task[] initializeTaskList() {
+        return new Task[MAX_NO_OF_TASKS];
+    }
+
+    private static int doTaskAndGetNewNumberOfTasks(Task[] taskList, int numberOfTasks, String userInput,
+                                                    String[] splitUserInput) {
+
+

```

Comment:
Would be nice to look through the layout to prevent blank lines etc and follow coding standard :smiley:

Reply:
I will see what I can do for the blank lines and long character lines  :+1: